### PR TITLE
add component test for istio eval profile with godog support

### PR DESCRIPTION
**Pull request by piotrkpc**
**Description**
Adds Istio component test for evaluation profile installation

Run the tests:
```bash
export KYMA_PROFILE=evaluation
make test-k3d # deply k3d, install kyma, run tests
or
make test # run tests (go test ./...)
```

**Related issue(s)**

https://github.com/kyma-project/kyma/issues/14256

### CODE REVIEW

**Review by barchw — APPROVED** [review]

**Comment by piotrkpc** [comment]
Notes with feedback from Oleksandr:
- the feature file does not captures full the intent and it could be improved. Example
```
Scenario: Istio component is installed with a configured profile
Istio component is always installed in a Kyma cluster with a
profile configured during an installation
  Given a running Kyma cluster
  And Kyma profile setting is "evaluation"
  Then Istio component is installed
  And Istio's profile is "evaluation"


Scenario: Istio component has all required pods running
  Given a running Kyma cluster with "evaluation" profile
  When Istio component is installed
  Then there is 1 pod for Pilot
  And there is 1 pod for "Ingress gateway"
  And "HPA" is not deployed
  And Istio pods are available for at least 3 seconds
```
- godog integration does not integrate with `testing.T` and relies on errors, this makes writing tests less readable but it can be improved in the future.

**Comment by mjakobczyk** [comment]
> Scenario: Istio component has **all required pods** running

What are **all required pods**? Should we define this term earlier or could it be more specific? From my point of view is too broad and should be narrowed down to for example: **internal components** or **control plane and date plane**.

>   Then there is 1 pod for Pilot
>   And there is 1 pod for "Ingress gateway"
>   And "HPA" is not deployed

Shouldn't we quote **Pilot** term in this scenario also? Or define some common vocabulary ahead to use them freely?


> And Istio pods are available for at least 3 seconds

Don't we specify startup availability such as after what time pods should be reachable?

**Comment by piotrkpc** [comment]
> Notes with feedback from Oleksandr:
> 
> * the feature file does not captures full the intent and it could be improved. Example
> 
> ```
> Scenario: Istio component is installed with a configured profile
> Istio component is always installed in a Kyma cluster with a
> profile configured during an installation
>   Given a running Kyma cluster
>   And Kyma profile setting is "evaluation"
>   Then Istio component is installed
>   And Istio's profile is "evaluation"
> 
> 
> Scenario: Istio component has all required pods running
>   Given a running Kyma cluster with "evaluation" profile
>   When Istio component is installed
>   Then there is 1 pod for Pilot
>   And there is 1 pod for "Ingress gateway"
>   And "HPA" is not deployed
>   And Istio pods are available for at least 3 seconds
> ```
> 
> * godog integration does not integrate with `testing.T` and relies on errors, this makes writing tests less readable but it can be improved in the future.

Since the first scenario is not really applicable since we don't have the setting of profile other than during deployment command.

Best practices I've tried to follow:
- keep the same order for registering steps with `ctx.Step` as we have in feature file.
Comments are welcome.

File: tests/components/istio/istio_installed_test.go [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+	}
+	k8sClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+	return k8sClient
+}
+
+func TestIstioInstalled(t *testing.T) {
+
```
Comment (cnvergence) [inline tests/components/istio/istio_installed_test.go:66]:
```suggestion

```

**Pull request merged**